test/cases/compile/require-context/index.js

```diff
@@ -0,0 +1,6 @@
+it("should maintain require context", function() {
+	var context = {foo: "bar"};
+	require(["./a", "./b"], function(a, b) {
+		this.foo.should.eql("bar");
+	}.bind(context));
```

Comment:
I think this doesn't tests your case. It must be `.bind(this)`. Best wrap the call in a IIFE to emulate the context.